SOQL Builder GA doc update (#3422)
forcedotcom/salesforcedx-vscode: docs/_articles/en/soql/writing.md

> VS Code supports code completions for SOQL `.soql` files (and will eventually support code completion in Apex files). To use this feature, you must refresh the SObject definitions so that the SOQL language server can provide code completion suggestions. Run `SFDX: Refresh SObject Definitions` from the Command Palette.

Reviewer: Do we want to say: (and we plan to eventually support code...) - so we don't make a concrete promise.

Reviewer 2: @sbudhirajadoc - The SOQL LSP supports code completion for SOQL embedded in Apex.  So this should be rephrased to remove the "(and will eventually..." piece and explain that the extensions support code completion for SOQL in a .soql file or when it's embedded in Apex.

> VS Code supports code completions for SOQL `.soql` files (and will eventually support code completion in Apex files). To use this feature, you must refresh the SObject definitions so that the SOQL language server can provide code completion suggestions. Run `SFDX: Refresh SObject Definitions` from the Command Palette.

Reviewer: ```suggestion
VS Code supports code completions for SOQL `.soql` files (and plan to eventually support code completion in Apex files). To use this feature, refresh the SObject definitions so that the SOQL language server can provide code completion suggestions. Run `SFDX: Refresh SObject Definitions` from the Command Palette.
```

Author: @smaddox-sf OK to make this update?

Reviewer 2: @sbudhirajadoc - See comment above.  We already support code completion for embedded Apex.

- To navigate between the suggestions, use the arrow keys.
- To auto-complete from the suggestion, press Enter.

> 

Reviewer 2: @sbudhirajadoc - Below where we talk about method to execute queries, we should include SOQL Builder and link to that doc page.  The other options still exist, but feels incomplete to not include the new option with SOQL Builder.

Author: updated. @smaddox-sf
